ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -645,7 +669,32 @@ int
 wiredtiger_extension_init(WT_CONNECTION *connection, WT_CONFIG_ARG *config)
 {
     azure_store *azure_storage = new azure_store;
+    WT_CONFIG_ITEM v;
+
     azure_storage->wt_api = connection->get_extension_api(connection);
+    int ret = azure_storage->wt_api->config_get(
+      azure_storage->wt_api, nullptr, config, "verbose.tiered", &v);
+
+    azure_storage->verbose = WT_VERBOSE_ERROR;
+    // Initialise logger for the storage source.
+    if (ret == 0 && v.val >= WT_VERBOSE_ERROR && v.val <= WT_VERBOSE_DEBUG_5) {
+        azure_storage->verbose = v.val;
+        Logger::SetLevel(wt_to_azure_verbosity_level(v.val));
+    } else if (ret != WT_NOTFOUND) {
+        azure_storage->log->log_err_msg(
+          "wiredtiger_extension_init: error parsing config for verbose level.");
+        delete (azure_storage);
+        return (ret != 0 ? ret : EINVAL);
+    }
+
+    azure_storage->log =
+      std::make_unique<azure_log_system>(azure_storage->wt_api, azure_storage->verbose);
+    Logger::SetListener([azure_storage](auto lvl, auto msg) {
```

Comment:
Rather than creating an annoymous lambda function, I was thinking of creating a function itself and placing inside the azure_log_system.cpp class and passing that in? What do you think? The benefit is that it abstracts away the azure to wt verbosity and vice versa and placing that logic into the class itself.

Reply:
Not sure if this is what you were expecting but i've created a `set_listener` function called when the class is created that essentially does the same but now lives in `azure_log_system.cpp`. Please let me know what your thoughts are. https://github.com/wiredtiger/wiredtiger/pull/8834/commits/fd7497e8ac5783be0d87943eb82e6e22cc77ba4d

Reply:
Not sure if the comment was lost, but the idea was basically to have a function like:
```
Logger::SetListener(azure_log_listener);
```
and then 
```
azure_log_system::azure_log_listener(auto lvl, auto msg) {
           if (azure_to_wt_verbosity_level(lvl) <= _wt_verbosity_level)
              _wt_api->err_printf(_wt_api, NULL, "%s", msg.c_str());
          else
              _wt_api->msg_printf(_wt_api, NULL, "%s", msg.c_str());
}
```

Reply:
thanks for the help its now refactored in https://github.com/wiredtiger/wiredtiger/pull/8834/commits/1a85d68b179b622982fd10d057bfbeff6c9ad9a0